### PR TITLE
Update Config as Code documentation

### PR DESCRIPTION
**Pull request by shipra-octopus**
Update the documentation to include the following - 

- We have added 2 fields in build server integrations to specify branch variable name and commit.
- Provide some details on recommended scopes for the personal access token based on the least privilege principle.

### CODE REVIEW

**Comment by shipra-octopus** [comment]
Updated to include feedback from @zentron

**Review by zentron — APPROVED** [review]
Those changes look good to me 👍  nice one.

File: docs/projects/version-control/index.md [hunk 1]
```diff
@@ -81,3 +83,14 @@ When editing the deployment process via the Octopus UI, the branch is selected i
 ### Via code
 
 Changes can also be made using your favorite text-editor or IDE, and committed and pushed just as you would any other code change. 
+
+
+
+## Additional options in your Build Server integration
+
+Once an Octopus project is configured to be version-controlled, you can choose which branch to build from before creating a release in Octopus. To enable this, we have added the following two new fields to the our common integrations - TeamCity, Asure DevOps, Jenkins, Github Actions and Bamboo.
```
Comment (harrisonmeister) [inline docs/projects/version-control/index.md:91]:
```suggestion
Once an Octopus project is configured to be version-controlled, you can choose which branch to build from before creating a release in Octopus. To enable this, we have added the following two new fields to our common integrations - TeamCity, Azure DevOps, Jenkins, GitHub Actions, and Bamboo.
```

File: docs/projects/version-control/index.md [hunk 2]
```diff
@@ -81,3 +83,14 @@ When editing the deployment process via the Octopus UI, the branch is selected i
...
+
+* Git Reference - the user-friendly alias for a commit hash.
+* Git Commit - the commit SHA-1 hash.
+
+When the app being built is in a different repository to the octopus project, Octopus does not guess or auto-populate the commit or branch that you want to create the release from. Also, in the case where the app and the octopus project is in the same repository, the head of that branch could have moved forward from what is expected. So in both cases, it is highly recommended that you provide the commit and not just the branch.
```
Comment (harrisonmeister) [inline docs/projects/version-control/index.md:96]:
```suggestion
When the app being built is in a different repository to the Octopus project, Octopus does not guess or auto-populate the commit or branch that you want to create the release from. Also, in the case where the app and the Octopus project are in the same repository, the head of that branch could have moved forward from what is expected. In both cases, it is highly recommended that you provide the commit and not just the branch.
```

File: docs/projects/version-control/index.md [hunk 3]
```diff
@@ -57,14 +57,16 @@ The _Default Branch Name_ is the branch on which the Octopus configuration will
...
+The _Authentication_ field specifies the credentials used by Octopus when authenticating with the git provider.  For the Password field, we recommend using a personal access token. We also recommend that you follow the principle of least privilege when selecting scopes or permissions to grant this personal access token. 
 
-* [GitHub - Creating a personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token)
-* [Azure DevOps](https://docs.microsoft.com/en-us/azure/devops/organizations/accounts/use-personal-access-tokens-to-authenticate)
-* [BitBucket](https://confluence.atlassian.com/bitbucketserver063/personal-access-tokens-972354166.html)
-* [GitLab](https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html)
+Git providers allow you to create an access token in different ways. The recommended *scope* for each provider is also listed in brackets. 
+
+* [GitHub - Creating a personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token); (Scope - *repo*)
```
Comment (harrisonmeister) [inline docs/projects/version-control/index.md:66]:
```suggestion
* [GitHub - Creating a personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token); (Scope - `repo`)
```

File: docs/projects/version-control/index.md [hunk 4]
```diff
@@ -57,14 +57,16 @@ The _Default Branch Name_ is the branch on which the Octopus configuration will
...
+* [Azure DevOps](https://docs.microsoft.com/en-us/azure/devops/organizations/accounts/use-personal-access-tokens-to-authenticate); (Scope - *vso.code_full*)
```
Comment (harrisonmeister) [inline docs/projects/version-control/index.md:67]:
```suggestion
* [Azure DevOps](https://docs.microsoft.com/en-us/azure/devops/organizations/accounts/use-personal-access-tokens-to-authenticate); (Scope - `vso.code_full`)
```

File: docs/projects/version-control/index.md [hunk 5]
```diff
@@ -57,14 +57,16 @@ The _Default Branch Name_ is the branch on which the Octopus configuration will
...
+* [BitBucket](https://confluence.atlassian.com/bitbucketserver063/personal-access-tokens-972354166.html); (Permission - *Project admin*)
```
Comment (harrisonmeister) [inline docs/projects/version-control/index.md:68]:
```suggestion
* [BitBucket](https://confluence.atlassian.com/bitbucketserver063/personal-access-tokens-972354166.html); (Permission - `Project admin`)
```

File: docs/projects/version-control/index.md [hunk 6]
```diff
@@ -57,14 +57,16 @@ The _Default Branch Name_ is the branch on which the Octopus configuration will
...
+* [GitLab](https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html); (Scope - *write_repository*)
```
Comment (harrisonmeister) [inline docs/projects/version-control/index.md:69]:
```suggestion
* [GitLab](https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html); (Scope - `write_repository`)
```

**Review by harrisonmeister — APPROVED** [review]
LGTM (made minor formatting and grammar changes)

**Pull request merged**